src/evaluators/gather/PHAL_GatherSolution_Def.hpp

```diff
@@ -138,10 +138,16 @@ GatherSolutionBase(const Teuchos::ParameterList& p,
 
   if ( tensorRank == 0 ) {
     device_sol.val_kokkos.resize(numFields);
+    device_sol.val_kokkos.sync_host();
+    device_sol.val_kokkos.sync_device();
```

Comment:
Why do you need to sync twice? Isn't `sync_host` doing "copy device to host", and hence ending with a fully synced view?

Reply:
If we could guarantee that the result of `resize` sets the modify flag on the device side, that would be all that is needed.

I looked at the implementation (https://github.com/kokkos/kokkos/blob/master/containers/src/Kokkos_DualView.hpp#L1092) and it seems like it is possible that in certain configurations, the modify flag could get set on host instead. Although maybe in that case, there is no device to sync to?

Reply:
Ok, that's fine. This is not that crucial anyways.